### PR TITLE
docs(troubleshoot): Add docker error troubleshooting guide

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Tracking issue
N/A

## Why are the changes needed?

Some users like me encountered the docker error while trying to start Flyte.

Untitled

## What changes were proposed in this pull request?

Add troubleshooting guide for docker error.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

**Review by pingsutw — DISMISSED** [review]
Thank you!

File: docs/community/troubleshoot.rst [hunk 1]
```diff
@@ -32,6 +32,31 @@ Depending on the contents of the logs or the `Events`, you can try different thi
 Debugging common execution errors
 ----------------------------------
 
+``Error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This error will show if your are not running Docker with the native Docker engine in a Linux machine. Most probably you are running Docker via Docker Desktop.
```
Comment (neverett) [inline docs/community/troubleshoot.rst:38]:
```suggestion
This error will show if you are not running Docker with the native Docker engine in a Linux machine. Most likely you are running Docker via Docker Desktop.
```

File: docs/community/troubleshoot.rst [hunk 2]
```diff
@@ -32,6 +32,31 @@ Depending on the contents of the logs or the `Events`, you can try different thi
...
+
+- If you are using Docker Desktop in MacOs, run
```
Comment (neverett) [inline docs/community/troubleshoot.rst:40]:
```suggestion
- If you are using Docker Desktop in MacOS, run:
```

File: docs/community/troubleshoot.rst [hunk 3]
```diff
@@ -32,6 +32,31 @@ Depending on the contents of the logs or the `Events`, you can try different thi
...
+
+.. prompt:: bash $
+
+ sudo ln -s ~/Library/Containers/com.docker.docker/Data/docker.raw.sock /var/run/docker.sock
+
+- If you are using Docker Desktop in Linux, run
```
Comment (neverett) [inline docs/community/troubleshoot.rst:46]:
```suggestion
- If you are using Docker Desktop in Linux, run:
```

File: docs/community/troubleshoot.rst [hunk 4]
```diff
@@ -32,6 +32,31 @@ Depending on the contents of the logs or the `Events`, you can try different thi
...
+
+.. prompt:: bash $
+
+ sudo ln -s ~$USER/.docker/desktop/docker.sock /var/run/docker.sock
+
+- If you are using some other tools to run Docker, you need to make sure that ``/var/run/docker.sock`` is linked to the correct socket file.
```
Comment (neverett) [inline docs/community/troubleshoot.rst:52]:
```suggestion
- If you are using another tool to run Docker, make sure that ``/var/run/docker.sock`` is linked to the correct socket file.
```

File: docs/community/troubleshoot.rst [hunk 5]
```diff
@@ -32,6 +32,31 @@ Depending on the contents of the logs or the `Events`, you can try different thi
...
+
+  For example, if you are using Rancher Desktop on Linux, run
```
Comment (neverett) [inline docs/community/troubleshoot.rst:54]:
```suggestion
  For example, if you are using Rancher Desktop on Linux, run:
```

**Review by neverett — CHANGES_REQUESTED** [review]
A few small changes requested, otherwise looks good. Thanks!

**Comment by MortalHappiness** [comment]
@neverett Fixed. Thank you.

**Review by pingsutw — APPROVED** [review]
Thank you @neverett @MortalHappiness

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Review by neverett — APPROVED** [review]

**Pull request merged**